x-pack/test/functional/apps/dataset_quality/dataset_quality_alerting.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DatasetQualityFtrProviderContext } from './config';
+import { getInitialTestLogs } from './data';
+import {
+  createDatasetQualityUserWithRole,
+  deleteDatasetQualityUserWithRole,
+} from './roles/role_management';
+
+export default function ({ getService, getPageObjects }: DatasetQualityFtrProviderContext) {
+  const PageObjects = getPageObjects([
+    'common',
+    'security',
+    'navigationalSearch',
+    'observabilityLogsExplorer',
+    'datasetQuality',
+  ]);
+  const security = getService('security');
+  const synthtrace = getService('logSynthtraceEsClient');
+  const testSubjects = getService('testSubjects');
+
+  const to = new Date(new Date().setDate(new Date().getDate() - 1)).toISOString();
+
+  describe('Dataset quality alerting', () => {
+    before(async () => {
+      await PageObjects.observabilityLogsExplorer.setupInitialIntegrations();
+    });
+
+    after(async () => {
+      await PageObjects.observabilityLogsExplorer.removeInstalledPackages();
+    });
+
+    describe('User cannot create rules', () => {
+      before(async () => {
+        // Index logs for synth-* and apache.access datasets
+        await synthtrace.index(getInitialTestLogs({ to, count: 4 }));
+
+        await createDatasetQualityUserWithRole(security, 'canManageAlerts');
+
+        // Logout in order to re-login with a different user
+        await PageObjects.security.forceLogout();
+        await PageObjects.security.login('canManageAlerts', 'canManageAlerts-password', {
+          expectSpaceSelector: false,
+        });
+
+        await PageObjects.datasetQuality.navigateTo();
+      });
+
+      after(async () => {
+        await synthtrace.clean();
+
+        // Cleanup the user and role
+        await PageObjects.security.forceLogout();
+        await deleteDatasetQualityUserWithRole(security, 'canManageAlerts');
+      });
+
+      it('page does not show create rule button', async () => {
+        await testSubjects.missingOrFail(
+          PageObjects.datasetQuality.testSubjectSelectors.datasetQualityDetailsHeaderButton
```

Comment:
Is it for a purpose that same value of `"datasetQualityDetailsHeaderButton"` for `data-test-subj` is used for both of the buttons?

https://github.com/elastic/kibana/blob/7b8a62dae2b5f36eab9f6e1631ca5f8bc772346d/x-pack/platform/plugins/shared/dataset_quality/public/components/dataset_quality/header.tsx#L86

https://github.com/elastic/kibana/blob/7b8a62dae2b5f36eab9f6e1631ca5f8bc772346d/x-pack/platform/plugins/shared/dataset_quality/public/components/dataset_quality_details/header.tsx#L160

Reply:
Maybe I'm missing something, but I don't see the problem in that since they are in different pages. By Url you can target the different buttons (e.g FullStory)